src/controllers/Index.php

```diff
@@ -91,18 +95,55 @@ public function home(Base $f3) {
         }
     }
 
+    /**
+     * Provide information about the selfoss instance.
+     * json
+     *
+     * @return void
+     */
+    public function about() {
+        $anonymizer = \helpers\Anonymizer::getAnonymizer();
+        $wallabag = !empty(\F3::get('wallabag')) ? [
+            'url' => \F3::get('wallabag'), // string
+            'version' => \F3::get('wallabag_version'), // int
+        ] : null;
+
+        $configuration = [
+            'version' => \F3::get('version'),
+            'apiversion' => \F3::get('apiversion'),
+            'configuration' => [
+                'homepage' => \F3::get('homepage') ? \F3::get('homepage') : 'newest', // string
+                'anonymizer' => $anonymizer === '' ? null : $anonymizer, // ?string
+                'share' => (string) \F3::get('share'), // string
+                'wallabag' => $wallabag, // ?array
+                'wordpress' => \F3::get('wordpress'), // ?string
+                'autoMarkAsRead' => \F3::get('auto_mark_as_read') == 1, // bool
+                'autoCollapse' => \F3::get('auto_collapse') == 1, // bool
+                'autoStreamMore' => \F3::get('auto_stream_more') == 1, // bool
+                'loadImagesOnMobile' => \F3::get('load_images_on_mobile') == 1, // bool
+                'itemsPerPage' => \F3::get('items_perpage'), // int
+                'unreadOrder' => \F3::get('unread_order'), // string
+                'autoHideReadOnMobile' => \F3::get('auto_hide_read_on_mobile') == 1, // bool
+                'scrollToArticleHeader' => \F3::get('scroll_to_article_header') == 1, // bool
+                'htmlTitle' => trim(\F3::get('html_title')), // string
+                'allowPublicUpdate' => \F3::get('allow_public_update_access') == 1, // bool
+                'publicMode' => \F3::get('public') == 1, // bool
+                'authEnabled' => \F3::get('auth')->enabled() === true, // bool
+                'language' => \F3::get('language'), // string
+            ],
+        ];
+
+        echo $this->view->jsonSuccess($configuration);
+    }
+
     /**
      * password hash generator
      * html
      *
      * @return void
      */
     public function password() {
-        $this->view->password = true;
-        if (isset($_POST['password'])) {
-            $this->view->hash = hash('sha512', \F3::get('salt') . $_POST['password']);
-        }
-        echo $this->view->render('templates/hashpassword.phtml');
+        echo file_get_contents(BASEDIR . '/public/hashpassword.html');
```

Comment:
Maybe remove this function and serve the static file from the webserver using rewriting?

Reply:
I started with `/` and I initially wanted to do this but the fact that we use `/` for both the page and the API endpoint prevented me from doing this. Then I did not re-evaluate it for the other pages. But since `GET /` is not actually a part of public API, `GET /items` is, we probably could get away with moving it.

Reply:
One benefit of sending them through PHP is that we do not have to disable caching in the web server. To not delay this any further, I decided not to keep the current behaviour for now.